internal/storage/blob/store.go

```diff
@@ -350,21 +460,38 @@ func (s *Store) RepoStats(ctx context.Context) storage.RepoStats {
 }
 
 func (s *Store) Reload(ctx context.Context) error {
-	changes, err := s.clone(ctx)
+	cr, err := s.clone(ctx)
 	if err != nil {
-		return fmt.Errorf("failed to clone: %w", err)
+		return fmt.Errorf("failed to clone blob store: %w", err)
 	}
 
-	if failures := changes.failures(); failures > 0 {
-		logging.ReqScopeLog(ctx).Warn(fmt.Sprintf("Failed to download (%d) files", failures))
+	if cr.failuresCount > 0 {
+		logging.ReqScopeLog(ctx).Warn(fmt.Sprintf("Failed to download (%d) files", cr.failuresCount))
```

Comment:
I think we should return an error here and exit because the download failures are quite likely to make the index invalid anyway.

Reply:
Now that I think about it, I think I will get rid of `failuresCount` all together.
If there are any type of error during clone, we should abort to keep things atomic hence we do not require any state for failures IMO.